core/src/main/java/com/linecorp/armeria/server/ServerPort.java

```diff
@@ -66,14 +92,17 @@ public ServerPort(InetSocketAddress localAddress, SessionProtocol protocol) {
             }
         }
 
-        requireNonNull(protocol, "protocol");
-        checkArgument(protocol == HTTP || protocol == HTTPS,
-                      "protocol: %s (expected: %s or %s)", protocol, HTTP, HTTPS);
-
+        requireNonNull(protocols, "protocols");
+        checkArgument(protocols.stream().allMatch(p -> p == HTTP || p == HTTPS || p == PROXY),
+                      "protocol: %s (expected: %s, %s or %s)", protocols, HTTP, HTTPS, PROXY);
         this.localAddress = localAddress;
-        this.protocol = protocol;
+        this.protocols = ImmutableSortedSet.copyOf(protocols);
+
+        final StringJoiner protocolNameJoiner = new StringJoiner("+");
+        this.protocols.forEach(p -> protocolNameJoiner.add(p.uriText()));
```

Comment:
Can use `.stream()` (may want a utility method for this, a static method on `SessionProtocol` seems reasonable)

Reply:
Didn't need to use string joiner at all in this case. Will just use `Set.toString()`.